doc/static-route/SONiC_static_route_bfd_hld.md

```diff
@@ -0,0 +1,300 @@
+        * Delete this nexthop from the static route entry's nexthop list 
+            * if the static route entry's nexthop list is empty, delete this static route from redis appl_db, break;
+            * if the static route entry's nexthop list is NOT empty, write it to redis appl_db STATIC_ROUTE_TABLE with "expiry"="false".
+<br>
+<br>
+
+## Table reconciliation after StaticRouteBfd crash/restart
+When StaticRouteBfd crashes or restarts, it loses all the internal table contents. Need to rebuild the internal tables from redis DB (i.e., configuration DB, application DB, state DB, etc), and do cross checking between these tables and redis db to make the information consistent among these tables, and then start to process the events received from redis DB.
+
+### Start event listening
+Start event listening before the following table and redis DB reconciliation, so the event won't be missing during the reconciliation.
+<br>
+
+### Build TABLE_CONFIG
+Read redis config_db STATIC_ROUTE_TABLE, pick the route with "bfd"="true" static route and build TABLE_CONFIG
+<br>
+
+### Build TABLE_NEXTHOP
+Loop each entry in TABLE_CONFIG, build TABLE_NEXTHOP
+<br>
+
+### Build TABLE_BFD and create BFD session if needed
+* 1\. read redis appl_db BFD_SESSION_TABLE, lookup the above TABLE_NEXTHOP, build internal TABLE_BFD table.<br>
+      *Notes: there is a bfd session leak risk here: when the last crash point is after deleting a nexthop (caused by deleting a static route in config) but before deleting BFD session from redis DB. The recover procedure cannot tell if this BFD session was created from StaticRouteBfd but no nexthop need it anymore.*
+
+* 2\. loop TABLE_NEXTHOP, create BFD session for the nexthop which has no corresponding entry in TABLE_BFD
+<br>
+
+### Build TABLE_SR and sync-up with redis appl_db STATIC_ROUTE_TABLE
+* 1\. loop TABLE_CONFIG to build TABLE_SR with empty nexthop list
+* 2\. read redis state_db BFD_SESSION_TABLE, for each BFD session in TABLE_BFD and the BFD session state is UP, get nexthop
+    * Get prefix list from TABLE_NEXTHOP, and loop this list. For each prefix lookup TABLE_SR to get prefix entry
+        * Add the above nexthop to this prefix's nexthop list
+* 3\. read redis appl_db STATIC_ROUTE_TABLE, collect all the entries with "expiry"="false" (StaticRouteBfd created static route entry)
+* 4\. loop TABLE_SR table, for each static route, compare with the route in above step #3, 
+    * Skip this entry if the static route matches (same nexthop list)
+    * Delete the redis appl_db static route if the nexthop list is empty in TABLE_SR
+    * Update static route in redis appl_db, "expiry"="false".
+<br>
+
+### Start event processing
+Start event processing after the above table and redis DB reconciliation.<br>
+In the rare case that, data were read from DB but also get an event later, cause rewrite BFD session or Static Route to appl_DB with same contents should not be a problem.
```

Comment:
Can add sections for the new tables schema?

Reply:
Is it better to create a PR to update the schema on the original static route document?
https://github.com/sonic-net/SONiC/blob/master/doc/static-route/SONiC_static_route_hdl.md#3211-static_route
There is a link to the above document. 
if the schema in two different documents, need to update both documents to keep them in sync.